federationapi/consumers/roomserver.go

```diff
@@ -146,28 +146,33 @@ func (s *OutputRoomEventConsumer) processInboundPeek(orp api.OutputNewInboundPee
 // processMessage updates the list of currently joined hosts in the room
 // and then sends the event to the hosts that were joined before the event.
 func (s *OutputRoomEventConsumer) processMessage(ore api.OutputNewRoomEvent) error {
-	eventsRes := &api.QueryEventsByIDResponse{}
-	if len(ore.AddsStateEventIDs) > 0 {
+	addsStateEvents := make([]*gomatrixserverlib.HeaderedEvent, 0, len(ore.AddsStateEventIDs))
+	missingEventIDs := make([]string, 0, len(ore.AddsStateEventIDs))
+	for _, eventID := range ore.AddsStateEventIDs {
```

Comment:
This is basically identical to syncapi. It would be nicer if `OutputNewRoomEvent` had a function on it to return `addsStateEvents, missingEventIDs` perhaps, then both fedapi and syncapi can just call that and then hit roomserver/local db.